engines/native/sandbox.go

```diff
@@ -35,6 +38,13 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 		return nil, fmt.Errorf("Failed to temporary folder, error: %s", err)
 	}
 
+	if b.payload.Context != "" {
+		if err = fetchContext(b.payload.Context, homeFolder.Path()); err != nil {
+			b.context.LogError(err)
+			return nil, err
```

Comment:
If there is a problem with the payload you should return a `MalformedPayloadError` not arbitrary errors... We can't gracefully handling arbitrary errors.